### DISCUSSION TITLE
Send wire format to the browser?

### DISCUSSION THREAD

**Discussion by JAForbes** [porsager/postgres#619] [Ideas]
Is it possible to send the binary wire format to the browser and decode it there?  I'm thinking this could be a great alternative to say AVRO or ProtoBuf

Is the decoding of the wire format something that could run client side?

**Comment by porsager** [comment]
Yeah, it's probably even closer after the cloudflare support. I'm not sure there would be much benefit, since the startup is quite chatty, same reason usage in cf doesn't make much sense in the bigger picture.

**Comment by M-jerez** [comment]
HI @porsager I'm investigating a bit more on this. 
**So you think sending wire format/binary to the browser won't be a big benefit?** 
 Avoiding all conversion to JS objects and then again to convert those to JSON.
 
 
On top of that i was also  looking into the option to using a Proxy to access data pg binary format instead having to parse all, just in case the server still needs to access the data. 
Would all this be possible (and ofc beneficial) or Am I just smoking too much 🌿😂?
here is my conversation with CHAT-GPT about it [(bottom part of the conversation)](https://chatgpt.com/share/67e6a008-1014-8002-9727-e653cafc50cc) 😂
I'm the author of [mion.io](https://github.com/MionKit/mion) and investigating a binary format for responses so thought that maybe using PG binary format would be straight forward better that protocol buffers or similar so there is no need for conversions.
If something good can come up from this I'm willing to work on this as well
